### PR TITLE
Fix issue with instant on session cache

### PR DESCRIPTION
**Pull request by wrodrig**
It seems that #28216 introduced lambda function inside the `CachingProviderService.java`. This 
 broke the `JCacheDynamicUpdateTest.dynamicUpdate`  because the FFDC instrumentation stopped inserting the expected ffdc.  In this PR we are addressing this failure. 

cc @tjwatson

### CODE REVIEW

**Comment by wrodrig** [comment]
#build
#Libby

**Review by jhanders34 — APPROVED** [review]

**Comment by wrodrig** [comment]
#build
#spawn.full.fat=io.openliberty.jcache.internal_fat.hazelcast,io.openliberty.jcache.internal_fat.infinispan

**Comment by wrodrig** [comment]
#build
#spawn.fullfat.buckets=io.openliberty.jcache.internal_fat.hazelcast,io.openliberty.jcache.internal_fat.infinispan

File: dev/io.openliberty.jcache.internal/src/io/openliberty/jcache/internal/CachingProviderService.java [hunk 1]
```diff
@@ -81,19 +83,23 @@ public void activate(Map<String, Object> configProps) throws Exception {
...
          * running with SecurityManager, permissions will need to be granted explicitly.
          */
-        CheckpointPhase.onRestore(0, () -> {
-            try {
-                ClassLoader classloader = getUnifiedClassLoader();
-                if (cachingProviderClass != null && !cachingProviderClass.trim().isEmpty()) {
-                    cachingProvider = Caching.getCachingProvider(cachingProviderClass, classloader);
-                } else {
-                    cachingProvider = Caching.getCachingProvider(classloader);
+        CheckpointPhase.onRestore(0, new OnRestore() {
```
Comment (tjwatson) [inline dev/io.openliberty.jcache.internal/src/io/openliberty/jcache/internal/CachingProviderService.java:86]: To make sure I understand, so all lambda's that have a catch will not have any FFDC processing?  I am curious how you found this issue?

**Review by tjwatson — APPROVED** [review]

File: dev/io.openliberty.jcache.internal/src/io/openliberty/jcache/internal/CachingProviderService.java (hunk 1 above)
Comment (wrodrig) [reply dev/io.openliberty.jcache.internal/src/io/openliberty/jcache/internal/CachingProviderService.java:86]: Thanks for your response! @jhanders34  actually found the root cause of the problem. In our case we were seeing consistent failures in `JCacheDynamicUpdateTest.dynamicUpdate`

Comment (tjwatson) [reply dev/io.openliberty.jcache.internal/src/io/openliberty/jcache/internal/CachingProviderService.java:86]: Thanks, that was a FULL test so explains why the PR builds allowed a merge.  Now I need to go review all the calls to onRestore that may have a similar issue.

Comment (jhanders34) [reply dev/io.openliberty.jcache.internal/src/io/openliberty/jcache/internal/CachingProviderService.java:86]: Yeah I think we need to look at the instrumentation logic to see if a change needs to be made there for both FFDC and trace injection.

Comment (tjwatson) [reply dev/io.openliberty.jcache.internal/src/io/openliberty/jcache/internal/CachingProviderService.java:86]: I'm not sure trace is a big deal, unless you need to see trace into the entry of the lambda (`call` method here).  Some would argue the body of the lambda should be trivial and not need entry/exit tracing.  For example, maybe the body of this lamda should have been extracted into a method that the onRestore lambda would simply reference/call.

**Comment by wrodrig** [comment]
#build
#Libby

**Comment by wraschke** [comment]
#build
#Libby

**Comment by wraschke** [comment]
#libby

**Comment by wrodrig** [comment]
#build
#Libby

**Comment by wrodrig** [comment]
Failure is not related to my changes. Merging this one

**Pull request merged**